src/main/java/com/exasol/glue/filesystem/S3FileSystem.java

```diff
@@ -0,0 +1,140 @@
+package com.exasol.glue.filesystem;
+
+import java.io.Closeable;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import com.exasol.errorreporting.ExaError;
+import com.exasol.glue.ExasolOptions;
+import com.exasol.glue.S3ClientFactory;
+import com.exasol.glue.connection.ExasolConnectionException;
+
+import software.amazon.awssdk.core.exception.SdkClientException;
+import software.amazon.awssdk.services.s3.S3Client;
+import software.amazon.awssdk.services.s3.model.*;
+
+/**
+ * An S3 file system operations implementations.
+ */
+public final class S3FileSystem implements Closeable {
+    private static final Logger LOGGER = Logger.getLogger(S3FileSystem.class.getName());
+    private final S3Client s3Client;
+
+    /**
+     * Creates a new instance of {@link S3FileSystem}.
+     *
+     * @param options user provided options
+     */
+    public S3FileSystem(final ExasolOptions options) {
+        this.s3Client = new S3ClientFactory(options).getS3Client();
+    }
+
+    /**
+     * Creates a new instance of {@link S3FileSystem}.
+     *
+     * @param s3Client s3 client object
+     */
+    public S3FileSystem(final S3Client s3Client) {
+        this.s3Client = s3Client;
+    }
+
+    /**
+     * Checks if a given bucket exists.
+     *
+     * @param bucketName name of a bucket
+     * @return {@code true} if path is a directory, {@code false} otherwise
+     */
+    public boolean doesBucketExist(final String bucketName) {
+        try {
+            s3Client.headBucket(HeadBucketRequest.builder().bucket(bucketName).build());
+            return true;
+        } catch (final NoSuchBucketException exception) {
+            return false;
+        }
+    }
+
+    /**
+     * Deletes a given bucket contents.
+     *
+     * @param bucketName name of a bucket
+     */
+    public void deleteBucket(final String bucketName) {
+        LOGGER.info(() -> "Deleting S3 bucket '" + bucketName + "'.");
+        deleteObjects(bucketName, Optional.empty());
+    }
+
+    /**
+     * Deletes a given bucket with bucket key contents.
+     *
+     * @param bucketName name of a bucket
+     * @param bucketKey  bucket key value
+     */
+    public void deleteBucketKey(final String bucketName, final String bucketKey) {
```

Comment:
to me that name sounds like it would delete the bucket.
But it only deletes the contents right?

Reply:
Would `deleteBucketKeyContents` (similar other `deleteBucketContents`) be okay?

Reply:
how about `deleteKeys`?

Reply:
or `deleteObjects()`

Reply:
`deleteKeys` better